src/components/ContactsList/_modals/ContactUpsertModal/ContactUpsertModal.tsx

```diff
@@ -4,51 +4,74 @@ import { SrOnlySpan } from '@/styles';
 import { Loader } from '@/components/Loader';
 import { PaymailInput } from '@/components/Input/PaymailInput';
 import { Input } from '@/components/Input';
-import { addContact } from '@/api/requests/contact';
+import { upsertContact } from '@/api/requests/contact';
 import { modalCloseTimeout } from '@/components/Modal/modalCloseTimeout';
 import { ErrorBar } from '@/components/ErrorBar';
+import { isValidPhone } from '@/utils/helpers/validatePhone';
+import { ContactFields } from './useContactFields';
 
-type ContactAddModalProps = {
-  open: boolean;
+type ContactUpsertModal = {
   onSubmitted: () => void;
   onCancel: () => void;
+  sucessMsg: string;
+  errorMsg: string;
+  modalTitle: string;
+  modalSubtitle: string;
+  fields: ContactFields;
+  disabledPaymailInput: boolean;
 };
 
-export const ContactAddModal: FC<ContactAddModalProps> = ({ open, onSubmitted, onCancel }) => {
-  const [paymail, setPaymail] = useState<string>('');
-  const [name, setName] = useState<string>('');
+export const ContactUpsertModal: FC<ContactUpsertModal> = ({
+  onSubmitted,
+  onCancel,
+  fields,
+  errorMsg,
+  sucessMsg,
+  modalTitle,
+  modalSubtitle,
+  disabledPaymailInput,
+}) => {
   const [loading, setLoading] = useState(false);
-  const [error, setError] = useState(false);
-
+  const [error, setError] = useState<string | undefined>(undefined);
   const [successMessage, setSuccessMessage] = useState<string>('');
+  const { name, paymail, phone, setName, setPaymail, setPhone } = fields;
 
   const onSuccess = async () => {
-    setSuccessMessage('Contact added successfully!');
+    setSuccessMessage(sucessMsg);
     await modalCloseTimeout();
     onSubmitted();
   };
 
   const onSubmit = async () => {
+    if (phone && !isValidPhone(phone)) {
+      setError('Wrong phone number.');
+      return;
+    }
+    if (!paymail || !name) {
+      setError('Both Paymail and Name are required');
+      return;
+    }
     setLoading(true);
-    setError(false);
+    setError(undefined);
```

Comment:
Just a question. Isn't it a good practice to not use `undefined` and use `null` instead?

Reply:
Just a reply 😄  Here I should have used `null` I think. I'll change this.